ldk/javascript/examples/self-test-loop/src/tests/whisper/whisper-update.ts

```diff
@@ -377,3 +379,123 @@ export const testWhisperStateOnChange = (): Promise<boolean> =>
       // do nothing.
     }
   });
+
+export const testNonTextInputs = (): Promise<boolean> =>
+  new Promise(async (resolve, reject) => {
+    const selectValue = 1;
+
+    const checkbox: Checkbox = {
+      label: 'Check this box',
+      key: 'Checkbox',
+      onChange: () => {
+        // do nothing.
+      },
+      id: 'myCheckbox',
+      type: WhisperComponentType.Checkbox,
+    };
+
+    const radioButton: RadioGroup = {
+      type: WhisperComponentType.RadioGroup,
+      id: 'radioInputId',
+      key: 'radioInputId',
+      options: ['Select this option', 'Select other option'],
+      onSelect: () => {
+        // do nothing.
+      },
+    };
+
+    const selectInput: Select = {
+      type: WhisperComponentType.Select,
+      label: 'Select blue',
+      key: 'Select',
+      options: ['red', 'blue'],
+      onSelect: (error, param, incomingWhisper) => {
+        // do nothing
+      },
+    };
+
+    const firstConfirm = createConfirmOrDenyWithPromise(
+      'Interact with each component and click yes',
+      'User clicked no',
+    );
+    const testWhisper = await whisper.create({
+      label: 'Test Non Text Inputs State Update',
+      onClose: () => {
+        // do nothing.
+      },
+      components: [checkbox, radioButton, selectInput, ...firstConfirm.button],
+    });
+    await firstConfirm.promise;
+    const secondUpdate = createConfirmOrDenyWithPromise(
+      'Did the components maintain their value?',
+      'User selected update failed.',
+    );
+    await testWhisper.update({
+      label: 'Updated Whisper label :o',
+      components: [checkbox, radioButton, selectInput, ...secondUpdate.button],
+    });
+    await secondUpdate.promise;
+    resolve(true);
+  });
+
+export const testNonTextInputsWithValue = (): Promise<boolean> =>
+  new Promise(async (resolve, reject) => {
+    const selectValue = 1;
```

Comment:
you may want to wrap all await functions with try/catch to catch any exceptions and resolve/reject gracefully.